src/Azure.Functions.Cli/Kubernetes/KubernetesHelpers.cs

```diff
@@ -256,11 +282,97 @@ internal static string GetKedaResources(string @namespace)
+                else if (funcKeysToRemove?.Any() == true)
+                {
+                    foreach (var item in funcKeysToRemove)
+                    {
+                        allExistingFuncKeys.Data.Remove(item);
+                    }
+
+                    kubernetesKeysSecretResource = allExistingFuncKeys;
+                }
+                //If all existing keys needs to remain unchaged
+                else if (unchangedFuncKeys?.Any() == true)
+                {
+                    kubernetesKeysSecretResource = allExistingFuncKeys;
+                }
+                //All keys are the new keys
+                else
+                {
+                    kubernetesKeysSecretResource = GetSecret(keysSecretCollectionName, @namespace, currentImageFuncKeys);
+                    newFuncKeys = kubernetesKeysSecretResource.Data;
+                }
+
+                result.Insert(resourceIndex, kubernetesKeysSecretResource);
+                resourceIndex++;
 
-            return scaledobject != null
-                ? result.Append(scaledobject)
-                : result;
+                //if function keys Secrets needs to be mounted as volume in the function runtime container
+                if (mountKeysAsContainerVolume)
+                {
+                    FuncAppKeysHelper.CreateFuncAppKeysVolumeMountDeploymentResource(deployments, keysSecretCollectionName);
+                }
+                //Create the Pod identity with the role to modify the function kubernetes secret
+                else
+                {
+                    var svcActName = $"{name}-function-keys-identity-svc-act";
+                    var svcActDeploymentResource = GetServiceAccount(svcActName, @namespace);
+                    result.Insert(resourceIndex, svcActDeploymentResource);
+                    resourceIndex++;
+
+                    var funcKeysManagerRoleName = "functions-keys-manager-role";
+                    var secretManagerRole = GetSecretManagerRole(funcKeysManagerRoleName, @namespace);
+                    result.Insert(resourceIndex, secretManagerRole);
+                    resourceIndex++;
+
+                    var roleBindingName = $"{svcActName}-functions-keys-manager-rolebinding";
+                    var funcKeysRoleBindingDeploymentResource = GetRoleBinding(roleBindingName, @namespace, funcKeysManagerRoleName, svcActName);
+                    result.Insert(resourceIndex, funcKeysRoleBindingDeploymentResource);
+                    resourceIndex++;
+
+                    //add service account identity to the pod
+                    foreach (var deployment in deployments)
+                    {
+                        deployment.Spec.Template.Spec.ServiceAccountName = svcActName;
+                    }
+                }
+            }
+
+            result = result.Concat(deployments).ToList();
+            return (scaledobject != null ? result.Append(scaledobject) : result, unchangedFuncKeys, newFuncKeys);
```

Comment:
merge `unchangedFuncKeys` and `newFuncKeys` to simplify `GetFunctionsDeploymentResources()` contract